### ISSUE TITLE
When ImplicitlyValidateChildProperties is true, then set AutomaticValidationEnabled = false is invalid

### ISSUE THREAD

**Issue by xiaoxiaotank** [FluentValidation/FluentValidation#1868]
### FluentValidation version

10.3.5

### ASP.NET version

.NET 5

### Summary

I found a problem. It may be a bug.

when I set `AutomaticValidationEnabled = false` and `.ImplicitlyValidateChildProperties = true`, the `AutomaticValidationEnabled ` is not work.it still executes automatically.

### Steps to Reproduce

Please review the following code:
```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddControllersWithViews()
	.AddFluentValidation(fv =>
       {
	    fv.RegisterValidatorsFromAssemblyContaining<Startup>();
	    fv.ImplicitlyValidateChildProperties = true;
	    fv.AutomaticValidationEnabled = false;
	});
}
```
I expect `AutomaticValidationEnabled` effectively prevents automatic validation, but it still executes automatically.

For example:
```csharp
[HttpPost]
public string Post([FromBody] CreateUserDto input)
{
    if (!ModelState.IsValid) 
    {
	return "Model State："
	    + string.Join(Environment.NewLine,
		ModelState.Values.SelectMany(v => v.Errors.Select(e => e.ErrorMessage)));
    }

    return JsonSerializer.Serialize(input);
}

public class CreateUserDto 
{
    public int Age { get; set; }
}

public class CreateUserDtoValidator : AbstractValidator<CreateUserDto>
{
    public CreateUserDtoValidator() 
    {
	RuleFor(x => x.Age).GreaterThan(0);
    }
}
```
then call it:
![PMU0U}1A`DC {7D}H H`L`O](https://user-images.githubusercontent.com/20281086/144594187-b8b70792-504e-4b14-a967-967ceb1d327d.png)

it means 'Age' must greater than '0'

I tried to find out why, maybe in `FluentValidationModelValidator.ShouldSkip` method, the code `if (rootMetadata == null) return true;`  need to Move outside?

**Comment by JeremySkinner** [comment]
Good catch, thanks. Fixed in 7865b14da98a726a9b1f559e41e2d85e84987902 and will be in the next release.

**Comment by JeremySkinner** [comment]
10.3.6 is now released with this change
